Mathlib/Data/ZMod/Basic.lean

```diff
@@ -1453,6 +1455,44 @@ instance instZModModule : Module (ZMod n) K :=
   Subtype.coe_injective.module _ (AddSubmonoidClass.subtype K) coe_zmod_smul
 
 end AddSubgroupClass
+
+variable (n)
+
+lemma ZModModule.char_nsmul_eq_zero (x : G) : n • x = 0 := by
```

Comment:
Do we need `ZModModule` as namespace? Would `ZMod` already work?
Can we even have these lemmas in the root namespace, for modules over rings of characteristic `n`?

Reply:
The situation here is that PFR had an ad hoc definition of "elementary `p`-groups" which is strictly less general and practical than `Module (ZMod p)` in that
* it only works for `p` a prime
* it doesn't contain the data of the `ZMod p` action

I deleted that definition in favor of the `Module (ZMod p)` spelling in https://github.com/teorth/pfr/commit/813d4991dd85950ba9bd0fcaeae4940614a4adbf. My reasoning is that this spelling is already extensively used in `Algebra.Module.ZMod` and that generalising to an arbitrary ring of characteristic `p`  would result in pain when all we want to state is that our group has characteristic dividing `p`. Furthermore, `CharP` is slightly too strong in that it requires the characteristic to be `p` on the nose, rather than merely dividing it.

Concretely, in your suggested lemma
```
lemma char_nsmul_eq_zero (R : Type*) (n : ℕ) [CharP n R] [Module R G] (x : G) : n • x = 0 := by
```
there is no way for Lean to figure `R` out from the resulting equality. And it would get boring to write `char_nsmul_eq_zero (ZMod n) n` every time.

Reply:
Fair enough. Could you please turn a version of your response into a module docstring for this section of the file?